### ISSUE TITLE
AWS::ServiceCatalog::CloudFormationProvisionedProduct.Outputs

### ISSUE THREAD

**Issue by JamieEdge** [aws-cloudformation/cfn-python-lint#1658]
In July 2020, support for retrieving the outputs of provisioned Service Catalog products [was added to CloudFormation](https://aws.amazon.com/blogs/mt/how-to-launch-secure-and-governed-aws-resources-with-aws-cloudformation-and-aws-service-catalog/).

Given the following example template:

```yaml
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  ProvisionedProduct:
    Type: AWS::ServiceCatalog::CloudFormationProvisionedProduct
    Properties:
      ProductName: example
      ProvisioningArtifactName: v1

Outputs:
  OutputValue:
    Value: !GetAtt ProvisionedProduct.Outputs.Example

  OutputSub:
    Value: !Sub '${ProvisionedProduct.Outputs.Example}-example'
```

The following linting errors are output when using cfn-lint 0.35.0:

```
E1010 Invalid GetAtt ProvisionedProduct.Outputs.Example for resource OutputValue
example.yml:12:5

E1019 Parameter ProvisionedProduct.Outputs.Example for Fn::Sub not found at Outputs/OutputSub/Value/Fn::Sub
example.yml:15:5
```

Access to these outputs using `!GetAtt` is documented in the announcement of this functionality (e.g. `!GetAtt ApacheALB.Outputs.ALBTarget`). Additionally the example template given above works correctly when deployed.

`Output` is documented in the [AWS::ServiceCatalog::CloudFormationProvisionedProduct return values](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-servicecatalog-cloudformationprovisionedproduct.html#aws-resource-servicecatalog-cloudformationprovisionedproduct-return-values) and appears to already be defined within cfn-lint CloudSpecs as a `Map` type.

I believe that this a false positive and am not aware of any alternative syntax to access these outputs (e.g. `!FindInMap` doesn't appear to be usable in this instance).

**Comment by kddejong** [comment]
Thanks for submitting this issue.

For my tracking this is the translated template.  This gets translated to string which masks the null to the CloudFormation service.  The cfn-lint ordering may be adjust to handle this situation correctly.  
```
   "DefinitionString": {
          "Fn::Join": [
            "\n",
            [
              "{",
              "    \"StartAt\": \"myState\",",
              "    \"States\": {",
              "        \"myState\": {",
              "            \"End\": true,",
              "            \"Resource\": \"${definition_substitution_1}\",",
              "            \"ResultPath\": null,",
              "            \"Type\": \"Task\"",
              "        }",
              "    }",
              "}"
            ]
          ]
        },
```
